### ISSUE TITLE
Invalid correction for Style/RedundantLineContinuation 

### ISSUE THREAD

**Issue by dmke** [rubocop/rubocop#11755]
## Expected behavior

The following spec should not offend the Style/RedundantLineContinuation cop:

```ruby
  it "quotes" do
    expect_escape("\"double\" and 'single' quotes").to eq \
      "\\glqq{}double\\grqq{} and \\glq{}single\\grq{} quotes"
  end
```

## Actual behavior

First observed here: https://github.com/digineo/texd-ruby/actions/runs/4593846656/jobs/8112193997

```console
bundle exec rubocop --safe --debug spec/lib/texd/helpers_spec.rb
For GEM_ROOT: configuration from GEM_ROOT/.rubocop.yml
configuration from GEM_ROOT/.gems/gems/rubocop-rails-2.18.0/config/default.yml
configuration from GEM_ROOT/.gems/gems/rubocop-rails-2.18.0/config/default.yml
Default configuration from GEM_ROOT/.gems/gems/rubocop-1.49.0/config/default.yml
configuration from GEM_ROOT/.gems/gems/rubocop-rake-0.6.0/config/default.yml
configuration from GEM_ROOT/.gems/gems/rubocop-rake-0.6.0/config/default.yml
configuration from GEM_ROOT/.gems/gems/rubocop-capybara-2.17.1/config/default.yml
configuration from GEM_ROOT/.gems/gems/rubocop-capybara-2.17.1/lib/../config/default.yml
configuration from GEM_ROOT/.gems/gems/rubocop-rspec-2.19.0/config/default.yml
configuration from GEM_ROOT/.gems/gems/rubocop-rspec-2.19.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning GEM_ROOT/spec/lib/texd/helpers_spec.rb
C

Offenses:

spec/lib/texd/helpers_spec.rb:19:59: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    expect_escape("\"double\" and 'single' quotes").to eq \ ...
                                                          ^
spec/lib/texd/helpers_spec.rb:24:75: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    expect_escape(%q("O'Neill", they shouted, "don't go 'there'".)).to eq \ ...
                                                                          ^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

(You can find the `.rubocop.yml` here: https://github.com/digineo/texd-ruby/blob/master/.rubocop.yml).

Auto-correcting would leave the aforementioned spec in this state:

```diff
diff --git a/spec/lib/texd/helpers_spec.rb b/spec/lib/texd/helpers_spec.rb
index 9031384..85d5c9a 100644
--- a/spec/lib/texd/helpers_spec.rb
+++ b/spec/lib/texd/helpers_spec.rb
@@ -16,13 +16,13 @@ RSpec.describe Texd::Helpers do
   end
 
   it "quotes" do
-    expect_escape("\"double\" and 'single' quotes").to eq \
-      "\\glqq{}double\\grqq{} and \\glq{}single\\grq{} quotes"
+    expect_escape("\"double\" and 'single' quotes").to eq
+    "\\glqq{}double\\grqq{} and \\glq{}single\\grq{} quotes"
   end
 
   it "doesn't quote egde case" do
-    expect_escape(%q("O'Neill", they shouted, "don't go 'there'".)).to eq \
-      "\\glqq{}O'Neill\\grqq{}, they shouted, \\glqq{}don't go \\glq{}there\\grq{}\\grqq{}."
+    expect_escape(%q("O'Neill", they shouted, "don't go 'there'".)).to eq
+    "\\glqq{}O'Neill\\grqq{}, they shouted, \\glqq{}don't go \\glq{}there\\grq{}\\grqq{}."
   end
```

... which isn't a valid invocation of `eq`:

```console
$ bundle exec rspec spec/lib/texd/helpers_spec.rb
...
Failures:

  1) Texd::Helpers quotes
     Failure/Error: expect_escape("\"double\" and 'single' quotes").to eq
     
     ArgumentError:
       wrong number of arguments (given 0, expected 1)
     # ./spec/lib/texd/helpers_spec.rb:19:in `block (2 levels) in <top (required)>'
```

## Steps to reproduce the problem

You'll need Ruby 2.7:

```console
$ git clone https://github.com/digineo/texd-ruby
$ cd texd-ruby
$ git checkout deps/rubocop
$ bundle install
$ bundle exec rake rubocop:autocorrect 
$ git diff spec/lib/texd/helpers_spec.rb
```

## RuboCop version

```
$ bundle exec rubocop -V
1.49.0 (using Parser 3.2.2.0, rubocop-ast 1.28.0, running on ruby 2.7.7) +server [x86_64-linux]
  - rubocop-rails 2.18.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.19.0
```
